android/java/org/chromium/chrome/browser/settings/BraveSearchEnginesPreferences.java

```diff
@@ -73,6 +83,49 @@ private void removePreferenceIfPresent(String key) {
         }
     }
 
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        updateCustomSearchEnginesPreference();
+    }
+
+    private void updateCustomSearchEnginesPreference() {
+        List<String> searchEngines = CustomSearchEnginesUtil.getCustomSearchEngines();
+
+        if (searchEngines == null || searchEngines.isEmpty()) {
+            removeCustomSearchEnginesPreference();
+            return;
+        }
+
+        PreferenceCategory preferenceCategory =
+                (PreferenceCategory) findPreference(PREF_CUSTOM_SEARCH_ENGINES_CATEGORY);
+        Preference customSearchEnginesListPreference =
```

Comment:
we definitely should not be duplicating information that is stored in the TemplateURLService and it's not clear to me why we need any of these custom search engine prefs

Reply:
Same reason as mentioned here : https://github.com/brave/brave-core/pull/27768#discussion_r2048376004
There's no built-in way to tell if a search engine was added by the user when using TemplateUrlService, so as a workaround, we save the keyword in prefs when the user adds it. Later on, we use that keyword to look up more details about the search engine through TemplateUrlService.